### PR TITLE
New Permission Model

### PR DESCRIPTION
**Pull request by SodhiA1**

### CODE REVIEW

File: backend/src/components/edx/exchange.js [hunk 1]
```diff
@@ -827,55 +863,60 @@ function getExchangeDocumentById() {
...
+      edxUserDistrictID: req.body.params.edxUserDistrictID,
+      edxUserExpiryDate: req.body.params.edxUserExpiryDate
+    };
+    await postData(token, config.get('server:edx:exchangeURL') + '/district-user-activation-relink-saga', payload,null, userName);
 
     return res.status(HttpStatus.OK).json('');
   } catch (e) {
-    log.error(e, 'removeUserSchoolAccess', 'Error occurred while attempting to remove user school access.');
+    log.error(e, 'relinkUserDistrictAccess', 'Error occurred while attempting to relink user school access.');
     return errorResponse(res);
```
Comment (SoLetsDev) [inline backend/src/components/edx/exchange.js:920]: small nitpik, I think it should say "relink user district access"

File: frontend/src/components/secure-message/AccessSchoolUsersPage.vue [hunk 2]
```diff
@@ -359,12 +359,17 @@ export default {
...
     getUsersData() {
       this.loadingUsers = true;
-      const payload = {params: {schoolID: this.schoolID}};
-      ApiService.apiAxios.get(Routes.edx.EXCHANGE_ACCESS_URL, payload)
+      ApiService.apiAxios.get(`${Routes.edx.EDX_SCHOOL_USERS_URL}/${this.schoolID}`)
         .then(response => {
           this.filteredUsers = this.sortUserData(response.data);
           this.users = this.filteredUsers;
-        }).finally(() => {
+          console.log(response)
```
Comment (SoLetsDev) [inline frontend/src/components/secure-message/AccessSchoolUsersPage.vue:366]: console.log

File: backend/src/components/utils.js [hunk 3]
```diff
@@ -180,6 +180,35 @@ async function logRequestData(operationType, url, data) {
   }
 }
 
+function checkUserHasPermission(permission) {
```
Comment (trev-dev) [inline backend/src/components/utils.js:184]: 👏 👏 👏

File: backend/src/util/Permission.js
Comment (trev-dev) [inline backend/src/util/Permission.js:1]: The strong enum pattern carries on in node :)

File: backend/src/components/edx/exchange.js
Comment (trev-dev) [inline backend/src/components/edx/exchange.js:1]: Separating concerns is a good thing!

**Review by arcshiftsolutions — APPROVED** [review]

**Pull request merged**